### PR TITLE
Frontend API implementations for Tasks

### PR DESCRIPTION
**Pull request by AXXPRO**
This pull request introduces several new APIs for task and user management, updates the `Task` and `User` models, and implements utility functions for formatting date-time strings. The most important changes are summarized below:

### New APIs for Task and User Management:
* Added multiple task-related APIs, including `saveTaskAPI`, `getAllTasksAPI`, `getTaskByIdAPI`, `deleteTaskByIdAPI`, `updateTaskByIdAPI`, `getTasksByPriorityAPI`, and `getTasksDoneAPI` in `farming_assistant/lib/APIs/task-related-apis.dart`.
* Added `loginAPI` for user authentication in `farming_assistant/lib/APIs/user-related-apis.dart`.

### Model Updates:
* Updated `Task` model to include new fields (`changeToMedium`, `changeToHigh`, `done`) and nullable types for existing fields. Added `toString` method for better logging in `farming_assistant/lib/models/task.dart`. [[1]](diffhunk://#diff-7ebe8ec79472b0c5f153ed49d8e368d6475f11a16eafa0668a39ab0a33ca002eR3-R50) [[2]](diffhunk://#diff-7ebe8ec79472b0c5f153ed49d8e368d6475f11a16eafa0668a39ab0a33ca002eL46-R74)
* Updated `User` model to use nullable types in `farming_assistant/lib/models/user.dart`.

### Utility Functions:
* Added `formatDateTimeString` utility function to format `DateTime` objects in `farming_assistant/lib/utils/date_time_formater.dart`.

### Enum Extensions:
* Added `jsonValue` extension to `LocationType`, `Priority`, `Recurrence`, and `Section` enums to get the uppercase string representation in their respective files. [[1]](diffhunk://#diff-2a5dfb601a20860284bc1ed7c0f99ebd442f4001d3d33404511d141d41218a1bR7-R12) [[2]](diffhunk://#diff-332856ace3240dd0feb5750123fa3209a23537f44c6525e38d51edf7a9fc43fcR6-R11) [[3]](diffhunk://#diff-3f7c6f86f2e2294ded4dec6aa8deae84b9b70be70077b88821b0cbf34c44631aR7-R12) [[4]](diffhunk://#diff-e1aed1b6f5bfbe1383204d83125c94e769c23ecc658db12d31fda4865ef15ef3L6-R12)

### Code Refactoring:
* Refactored `loginAPI` call in `farming_assistant/lib/screens/login_screen.dart` to use the new `loginAPI` method from `user-related-apis.dart`. [[1]](diffhunk://#diff-7875e135a6736bd6ab9f74cd91f660fe3c0f5909726d14ff425e18d698f05d1dL239-R239) [[2]](diffhunk://#diff-7875e135a6736bd6ab9f74cd91f660fe3c0f5909726d14ff425e18d698f05d1dL296-L316)

### CODE REVIEW

File: farming_assistant/lib/APIs/task-related-apis.dart [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+import 'package:http/http.dart' as http;
+
+import '../models/enums/priority.dart';
+import '../models/enums/recurrence.dart';
+import '../models/enums/sections.dart';
+import '../models/task.dart';
+import '../models/user.dart';
+import '../utils/config.dart';
+
+//PROVIDE EITHER THE ARGUMENTS OR THE TASK, NOT BOTH
```
Comment (Cozma-Alex) [inline farming_assistant/lib/APIs/task-related-apis.dart]: ce inseamna commul asta?

File: farming_assistant/lib/APIs/user-related-apis.dart [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+Future<User> loginAPI(User user) async {
+  final uri = Uri.parse('${APIConfig.baseURI}/users/auth');
+
+  try {
+    final response = await http.post(
+      uri,
+      headers: {
+        'Content-Type': 'application/json',
+        //Login does not require an authorization header
```
Comment (Cozma-Alex) [inline farming_assistant/lib/APIs/user-related-apis.dart]: no need for this comm(se intelege de la sine)

File: farming_assistant/lib/models/task.dart [hunk 3]
```diff
@@ -43,12 +56,20 @@ class Task {
...
+      'deadline':  formatDateTimeString(task.deadline!),
+      'change_to_medium': formatDateTimeString(task.changeToMedium!),
+      'change_to_high': formatDateTimeString(task.changeToHigh!),
+      'user': User.toJson(task.user!),
+      'done': task.done,
     };
   }
 
+  @override
+  String toString() {
```
Comment (Cozma-Alex) [inline farming_assistant/lib/models/task.dart:71]: de ce e fromJSON si toString?

File: farming_assistant/lib/models/user.dart [hunk 4]
```diff
@@ -1,23 +1,23 @@
...
-      jsonData['name'],
-      jsonData['image_data'],
+      id: jsonData['id'],
+      email: jsonData['email'],
+      password: jsonData['password_hash'],
+      farmName: jsonData['farm_name'],
+      name: jsonData['name'],
+      imageData: jsonData['image_data'],
     );
   }
```
Comment (Cozma-Alex) [inline farming_assistant/lib/models/user.dart:23]: aici ai si toJson si toString care fac more or less acelasi lucru, plus ca nu cred ca e nevoie sa trimiti obiectul pe care il faci JSON (valabil pt toate clasele de models)

**Review by Cozma-Alex — CHANGES_REQUESTED** [review]

File: farming_assistant/lib/models/task.dart (hunk 3 above)
Comment (AXXPRO) [reply farming_assistant/lib/models/task.dart:71]: toString-ul e doar pentru testare. 
Fara el o sa iti arate consola Instance of Task Object, care nu e prea usefull.
Sure, ii poti da task.toJson(), dar e mai intuitiv sa lasi doar obiectul

Comment (Cozma-Alex) [reply farming_assistant/lib/models/task.dart:71]: understandable, acuma cred ca e mai okk sa fie sters daca e doar pt testing si nu e folosit in alta parte

File: farming_assistant/lib/APIs/task-related-apis.dart (hunk 1 above)
Comment (AXXPRO) [reply farming_assistant/lib/APIs/task-related-apis.dart]: Outdated. Removed.

File: farming_assistant/lib/APIs/user-related-apis.dart (hunk 2 above)
Comment (AXXPRO) [reply farming_assistant/lib/APIs/user-related-apis.dart]: Outdated. Removed

**Review by Cozma-Alex — APPROVED** [review]

**Pull request merged**